## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#b4d49075-fb72-42cd-bc2a-def42bd5e5fa) |
| mint-fs.sh | :x: [more...](#7ea08f16-500f-418f-ad59-5858900b09b3) |
| mint-erasure.sh | :x: [more...](#56c9595e-e46e-4f57-8143-a051bc7f20ac) |
| mint-dist-erasure.sh | :x: [more...](#ee652dae-c9ba-4d54-be60-15967f2187b1) |
| mint-zoned.sh | :x: [more...](#6cbff881-ddf9-4d88-a3c8-2e03d0c6e1c0) |
| mint-gateway-nas.sh | :x: [more...](#fa6b65c1-8008-437f-b3d7-8911bcf267ab) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#9995c840-c8de-4fac-b0f0-a15e961029a0) |
<a name="fa6b65c1-8008-437f-b3d7-8911bcf267ab"></a>
### 12502-c9f1a61/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:31664
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 3d2b9348d89c:/mint/log /tmp/mint-logs'

(1/16) Running aws-sdk-go tests ... done in 0 seconds
(2/16) Running aws-sdk-java tests ... done in 1 seconds
(3/16) Running aws-sdk-php tests ... done in 41 seconds
(4/16) Running aws-sdk-ruby tests ... done in 2 seconds
(5/16) Running awscli tests ... FAILED in 51 seconds
{
  "name": "awscli",
  "duration": 2981,
  "function": "aws --endpoint-url http://minio-dev4.minio.io:31664 s3api get-bucket-lifecycle-configuration --bucket awscli-mint-test-bucket-748\n",
  "status": "FAIL",
  "error": "An error occurred (NotImplemented) when calling the GetBucketLifecycleConfiguration operation: A header you provided implies functionality that is not implemented"
}
(5/16) Running healthcheck tests ... done in 0 seconds
(6/16) Running mc tests ... done in 42 seconds
(7/16) Running minio-dotnet tests ... done in 27 seconds
(8/16) Running minio-go tests ... FAILED in 58 seconds
{
  "args": {
    "bucketName": "minio-go-test-54lml64hzl2guliu",
    "objectName": "093pw345mxykqa2eb3oljanbcns3w4",
    "sse": {}
  },
  "duration": 15,
  "error": "Server side encryption specified but KMS is not configured",
  "function": "PutEncryptedObject(bucketName, objectName, reader, sse)",
  "message": "PutEncryptedObject failed",
  "name": "minio-go: testSSES3EncryptionPutGet",
  "status": "FAIL"
}
(8/16) Running minio-java tests ... FAILED in 2 seconds
{
  "name": "minio-java",
  "function": "setBucketEncryption()",
  "duration": 33,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidArgument, message = Server side encryption specified but KMS is not configured, bucketName = minio-java-test-55po1p, objectName = null, resource = /minio-java-test-55po1p, requestId = 168876B744204B3E, hostId = f093adad-fed2-4fee-ae39-9f4f784d89e3)\nrequest={method=PUT, url=http://minio-dev4.minio.io:31664/minio-java-test-55po1p?encryption=, headers=Host: minio-dev4.minio.io:31664\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.2.1\nContent-MD5: 5bZurju84n5lUGMQ6mrGXQ==\nx-amz-content-sha256: df4e3e6cf7a0b4af67ba58f41eaa596d54579eea2ed2f993da09600f59413711\nx-amz-date: 20210614T134315Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210614/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 342\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 168876B744204B3E\nX-Xss-Protection: 1; mode=block\nDate: Mon, 14 Jun 2021 13:43:15 GMT\n}\n >>> [io.minio.S3Base.execute(S3Base.java:667), io.minio.S3Base.execute(S3Base.java:460), io.minio.S3Base.executePut(S3Base.java:794), io.minio.MinioClient.setBucketEncryption(MinioClient.java:2343), FunctionalTest.setBucketEncryption(FunctionalTest.java:3223), FunctionalTest.runBucketTests(FunctionalTest.java:3656), FunctionalTest.runTests(FunctionalTest.java:3719), FunctionalTest.main(FunctionalTest.java:3864)]"
}
(8/16) Running minio-js tests ... done in 1 minutes and 0 seconds
(9/16) Running minio-py tests ... done in 1 minutes and 15 seconds
(10/16) Running s3cmd tests ... done in 19 seconds
(11/16) Running s3select tests ... done in 2 seconds
(12/16) Running security tests ... done in 0 seconds
(13/16) Running versioning tests ... done in 0 seconds

Executed 13 out of 16 tests successfully.
```
<a name="6cbff881-ddf9-4d88-a3c8-2e03d0c6e1c0"></a>
### 12502-c9f1a61/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c2.minio.io:31013
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp a551a443f902:/mint/log /tmp/mint-logs'

(1/16) Running aws-sdk-go tests ... done in 1 seconds
(2/16) Running aws-sdk-java tests ... done in 2 seconds
(3/16) Running aws-sdk-php tests ... done in 43 seconds
(4/16) Running aws-sdk-ruby tests ... done in 4 seconds
(5/16) Running awscli tests ... done in 2 minutes and 20 seconds
(6/16) Running healthcheck tests ... done in 0 seconds
(7/16) Running mc tests ... done in 1 minutes and 15 seconds
(8/16) Running minio-dotnet tests ... done in 41 seconds
(9/16) Running minio-go tests ... FAILED in 1 minutes and 15 seconds
{
  "args": {
    "bucketName": "minio-go-test-cg1m1utws2nt3ib9",
    "objectName": "9pkmpsopg0uxux46lghn3fk9elxf66",
    "sse": {}
  },
  "duration": 68,
  "error": "Server side encryption specified but KMS is not configured",
  "function": "PutEncryptedObject(bucketName, objectName, reader, sse)",
  "message": "PutEncryptedObject failed",
  "name": "minio-go: testSSES3EncryptionPutGet",
  "status": "FAIL"
}
(9/16) Running minio-java tests ... FAILED in 4 seconds
{
  "name": "minio-java",
  "function": "setBucketEncryption()",
  "duration": 100,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidArgument, message = Server side encryption specified but KMS is not configured, bucketName = minio-java-test-160mq6e, objectName = null, resource = /minio-java-test-160mq6e, requestId = 168876DC7A643883, hostId = 3c5bba4f-e1be-441f-8bd1-81fcf6e7582a)\nrequest={method=PUT, url=http://minio-c2.minio.io:31013/minio-java-test-160mq6e?encryption=, headers=Host: minio-c2.minio.io:31013\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.2.1\nContent-MD5: 5bZurju84n5lUGMQ6mrGXQ==\nx-amz-content-sha256: df4e3e6cf7a0b4af67ba58f41eaa596d54579eea2ed2f993da09600f59413711\nx-amz-date: 20210614T134554Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210614/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 344\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 168876DC7A643883\nX-Xss-Protection: 1; mode=block\nDate: Mon, 14 Jun 2021 13:45:54 GMT\n}\n >>> [io.minio.S3Base.execute(S3Base.java:667), io.minio.S3Base.execute(S3Base.java:460), io.minio.S3Base.executePut(S3Base.java:794), io.minio.MinioClient.setBucketEncryption(MinioClient.java:2343), FunctionalTest.setBucketEncryption(FunctionalTest.java:3223), FunctionalTest.runBucketTests(FunctionalTest.java:3656), FunctionalTest.runTests(FunctionalTest.java:3719), FunctionalTest.main(FunctionalTest.java:3864)]"
}
(9/16) Running minio-js tests ... done in 53 seconds
(10/16) Running minio-py tests ... FAILED in 2 minutes and 13 seconds
{
  "name": "minio-py:test_thread_safe",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-61d00383-9058-4d02-bbc5-678659cfe78b",
    "object_name": "1a49df42-f382-40c3-b4b5-11e3c614ed6b"
  },
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /minio-py-test-61d00383-9058-4d02-bbc5-678659cfe78b, request_id: 16887707E4DC8E00, host_id: 3c5bba4f-e1be-441f-8bd1-81fcf6e7582a, bucket_name: minio-py-test-61d00383-9058-4d02-bbc5-678659cfe78b",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1640, in test_thread_safe\n    _CLIENT.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 690, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 422, in _execute\n    no_body_trace=no_body_trace,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 396, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /minio-py-test-61d00383-9058-4d02-bbc5-678659cfe78b, request_id: 16887707E4DC8E00, host_id: 3c5bba4f-e1be-441f-8bd1-81fcf6e7582a, bucket_name: minio-py-test-61d00383-9058-4d02-bbc5-678659cfe78b\n",
  "duration": 5044
}
(10/16) Running s3cmd tests ... done in 18 seconds
(11/16) Running s3select tests ... done in 6 seconds
(12/16) Running security tests ... done in 0 seconds
(13/16) Running versioning tests ... FAILED in 30 minutes and 6 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-qwh45enx4gjni1",
    "objectName": "testObject",
    "retentionMode": "COMPLIANCE"
  },
  "duration": 1801523,
  "function": "testPutGetRetentionCompliance",
  "message": "Unable to cleanup bucket after compliance tests",
  "name": "versioning",
  "status": "FAIL"
}

Executed 12 out of 16 tests successfully.
```
<a name="b4d49075-fb72-42cd-bc2a-def42bd5e5fa"></a>
### 12502-c9f1a61/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:31792
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 19941a17e95b:/mint/log /tmp/mint-logs'

(1/16) Running aws-sdk-go tests ... done in 2 seconds
(2/16) Running aws-sdk-java tests ... done in 1 seconds
(3/16) Running aws-sdk-php tests ... done in 43 seconds
(4/16) Running aws-sdk-ruby tests ... done in 3 seconds
(5/16) Running awscli tests ... done in 2 minutes and 23 seconds
(6/16) Running healthcheck tests ... done in 0 seconds
(7/16) Running mc tests ... done in 1 minutes and 12 seconds
(8/16) Running minio-dotnet tests ... done in 41 seconds
(9/16) Running minio-go tests ... FAILED in 1 minutes and 16 seconds
{
  "args": {
    "bucketName": "minio-go-test-jtlrwrgyhxkjcjit",
    "objectName": "dwytcbgxqdwvhi5fzcn6gsopscp6nf",
    "sse": {}
  },
  "duration": 66,
  "error": "Server side encryption specified but KMS is not configured",
  "function": "PutEncryptedObject(bucketName, objectName, reader, sse)",
  "message": "PutEncryptedObject failed",
  "name": "minio-go: testSSES3EncryptionPutGet",
  "status": "FAIL"
}
(9/16) Running minio-java tests ... FAILED in 4 seconds
{
  "name": "minio-java",
  "function": "setBucketEncryption()",
  "duration": 98,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidArgument, message = Server side encryption specified but KMS is not configured, bucketName = minio-java-test-16aa3p, objectName = null, resource = /minio-java-test-16aa3p, requestId = 168876DC83FB9C5E, hostId = 9eec9b69-652e-4f10-b157-2e5d23b76079)\nrequest={method=PUT, url=http://minio-dev4.minio.io:31792/minio-java-test-16aa3p?encryption=, headers=Host: minio-dev4.minio.io:31792\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.2.1\nContent-MD5: 5bZurju84n5lUGMQ6mrGXQ==\nx-amz-content-sha256: df4e3e6cf7a0b4af67ba58f41eaa596d54579eea2ed2f993da09600f59413711\nx-amz-date: 20210614T134555Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210614/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 342\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 168876DC83FB9C5E\nX-Xss-Protection: 1; mode=block\nDate: Mon, 14 Jun 2021 13:45:55 GMT\n}\n >>> [io.minio.S3Base.execute(S3Base.java:667), io.minio.S3Base.execute(S3Base.java:460), io.minio.S3Base.executePut(S3Base.java:794), io.minio.MinioClient.setBucketEncryption(MinioClient.java:2343), FunctionalTest.setBucketEncryption(FunctionalTest.java:3223), FunctionalTest.runBucketTests(FunctionalTest.java:3656), FunctionalTest.runTests(FunctionalTest.java:3719), FunctionalTest.main(FunctionalTest.java:3864)]"
}
(9/16) Running minio-js tests ... done in 53 seconds
(10/16) Running minio-py tests ... done in 2 minutes and 10 seconds
(11/16) Running s3cmd tests ... done in 20 seconds
(12/16) Running s3select tests ... done in 5 seconds
(13/16) Running security tests ... done in 0 seconds
(14/16) Running versioning tests ... done in 3 minutes and 6 seconds

Executed 14 out of 16 tests successfully.
```
<a name="9995c840-c8de-4fac-b0f0-a15e961029a0"></a>
### 12502-c9f1a61/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:30200
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 381e6df558fe:/mint/log /tmp/mint-logs'

(1/16) Running aws-sdk-go tests ... done in 1 seconds
(2/16) Running aws-sdk-java tests ... done in 2 seconds
(3/16) Running aws-sdk-php tests ... done in 43 seconds
(4/16) Running aws-sdk-ruby tests ... done in 3 seconds
(5/16) Running awscli tests ... FAILED in 32 seconds
{
  "name": "awscli",
  "duration": 2625,
  "function": "aws --endpoint-url http://minio-dev8.minio.io:30200 s3api copy-object --bucket awscli-mint-test-bucket-8118 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-8118/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected 29ad7016ed249036ac28ba4f0e893596, got fbaea6048f5e4bfb5ecdd8ed7395c5b3"
}
(5/16) Running healthcheck tests ... done in 0 seconds
(6/16) Running mc tests ... done in 53 seconds
(7/16) Running minio-dotnet tests ... done in 33 seconds
(8/16) Running minio-go tests ... FAILED in 1 minutes and 46 seconds
{
  "args": {
    "bucketName": "minio-go-test-epoenl6ah0apn1v4",
    "objectName": "yvy5xidwiyunezrwgrwaamp90tna4b",
    "sse": {}
  },
  "duration": 73,
  "error": "Server side encryption specified but KMS is not configured",
  "function": "PutEncryptedObject(bucketName, objectName, reader, sse)",
  "message": "PutEncryptedObject failed",
  "name": "minio-go: testSSES3EncryptionPutGet",
  "status": "FAIL"
}
(8/16) Running minio-java tests ... FAILED in 4 seconds
{
  "name": "minio-java",
  "function": "setBucketEncryption()",
  "duration": 118,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidArgument, message = Server side encryption specified but KMS is not configured, bucketName = minio-java-test-3bmbilj, objectName = null, resource = /minio-java-test-3bmbilj, requestId = 168876C393298FF0, hostId = 73a4c4db-723f-4477-97fc-16cb82bb4b11)\nrequest={method=PUT, url=http://minio-dev8.minio.io:30200/minio-java-test-3bmbilj?encryption=, headers=Host: minio-dev8.minio.io:30200\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.2.1\nContent-MD5: 5bZurju84n5lUGMQ6mrGXQ==\nx-amz-content-sha256: df4e3e6cf7a0b4af67ba58f41eaa596d54579eea2ed2f993da09600f59413711\nx-amz-date: 20210614T134408Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210614/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 344\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 168876C393298FF0\nX-Xss-Protection: 1; mode=block\nDate: Mon, 14 Jun 2021 13:44:08 GMT\n}\n >>> [io.minio.S3Base.execute(S3Base.java:667), io.minio.S3Base.execute(S3Base.java:460), io.minio.S3Base.executePut(S3Base.java:794), io.minio.MinioClient.setBucketEncryption(MinioClient.java:2343), FunctionalTest.setBucketEncryption(FunctionalTest.java:3223), FunctionalTest.runBucketTests(FunctionalTest.java:3656), FunctionalTest.runTests(FunctionalTest.java:3719), FunctionalTest.main(FunctionalTest.java:3864)]"
}
(8/16) Running minio-js tests ... done in 53 seconds
(9/16) Running minio-py tests ... done in 2 minutes and 41 seconds
(10/16) Running s3cmd tests ... FAILED in 5 seconds
{
  "name": "s3cmd",
  "duration": "2656",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "WARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Too many failures. Giving up on '/mint/data/datafile-65-MB'\nERROR: \nUpload of '/mint/data/datafile-65-MB' part 1 failed. Use\n  /usr/local/bin/s3cmd abortmp s3://s3cmd-test-bucket-32575/s3cmd-test-object-32657 b1389569-8f23-464f-83ee-4ee085d4ad78\nto abort the upload, or\n  /usr/local/bin/s3cmd --upload-id b1389569-8f23-464f-83ee-4ee085d4ad78 put ...\nto continue the upload.\nERROR: Upload of '/mint/data/datafile-65-MB' failed too many times (Last reason: )"
}
(10/16) Running s3select tests ... done in 5 seconds
(11/16) Running security tests ... done in 0 seconds
(12/16) Running versioning tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-hkn41fxaasypmw",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 209,
  "error": "unexpected ETag field",
  "function": "testPutObject",
  "message": "Unexpected list content",
  "name": "versioning",
  "status": "FAIL"
}

Executed 11 out of 16 tests successfully.
```
<a name="56c9595e-e46e-4f57-8143-a051bc7f20ac"></a>
### 12502-c9f1a61/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:30163
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp cdedae1f9449:/mint/log /tmp/mint-logs'

(1/16) Running aws-sdk-go tests ... done in 0 seconds
(2/16) Running aws-sdk-java tests ... done in 2 seconds
(3/16) Running aws-sdk-php tests ... done in 41 seconds
(4/16) Running aws-sdk-ruby tests ... done in 2 seconds
(5/16) Running awscli tests ... done in 2 minutes and 8 seconds
(6/16) Running healthcheck tests ... done in 0 seconds
(7/16) Running mc tests ... done in 1 minutes and 0 seconds
(8/16) Running minio-dotnet tests ... done in 32 seconds
(9/16) Running minio-go tests ... FAILED in 51 seconds
{
  "args": {
    "bucketName": "minio-go-test-klpgxsb9qen0lpz1",
    "objectName": "twrdupebrmb4swj4dxh0u6mbnat9be",
    "sse": {}
  },
  "duration": 46,
  "error": "Server side encryption specified but KMS is not configured",
  "function": "PutEncryptedObject(bucketName, objectName, reader, sse)",
  "message": "PutEncryptedObject failed",
  "name": "minio-go: testSSES3EncryptionPutGet",
  "status": "FAIL"
}
(9/16) Running minio-java tests ... FAILED in 2 seconds
{
  "name": "minio-java",
  "function": "setBucketEncryption()",
  "duration": 36,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidArgument, message = Server side encryption specified but KMS is not configured, bucketName = minio-java-test-2ti8jpc, objectName = null, resource = /minio-java-test-2ti8jpc, requestId = 168876CD20CB7E15, hostId = 8c6dc2d6-72e1-4934-8262-0e303e09e787)\nrequest={method=PUT, url=http://minio-dev8.minio.io:30163/minio-java-test-2ti8jpc?encryption=, headers=Host: minio-dev8.minio.io:30163\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.2.1\nContent-MD5: 5bZurju84n5lUGMQ6mrGXQ==\nx-amz-content-sha256: df4e3e6cf7a0b4af67ba58f41eaa596d54579eea2ed2f993da09600f59413711\nx-amz-date: 20210614T134449Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210614/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 344\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 168876CD20CB7E15\nX-Xss-Protection: 1; mode=block\nDate: Mon, 14 Jun 2021 13:44:49 GMT\n}\n >>> [io.minio.S3Base.execute(S3Base.java:667), io.minio.S3Base.execute(S3Base.java:460), io.minio.S3Base.executePut(S3Base.java:794), io.minio.MinioClient.setBucketEncryption(MinioClient.java:2343), FunctionalTest.setBucketEncryption(FunctionalTest.java:3223), FunctionalTest.runBucketTests(FunctionalTest.java:3656), FunctionalTest.runTests(FunctionalTest.java:3719), FunctionalTest.main(FunctionalTest.java:3864)]"
}
(9/16) Running minio-js tests ... done in 52 seconds
(10/16) Running minio-py tests ... done in 1 minutes and 28 seconds
(11/16) Running s3cmd tests ... done in 16 seconds
(12/16) Running s3select tests ... done in 2 seconds
(13/16) Running security tests ... done in 0 seconds
(14/16) Running versioning tests ... done in 3 minutes and 2 seconds

Executed 14 out of 16 tests successfully.
```
<a name="7ea08f16-500f-418f-ad59-5858900b09b3"></a>
### 12502-c9f1a61/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:30560
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 07eca01d9c76:/mint/log /tmp/mint-logs'

(1/16) Running aws-sdk-go tests ... done in 1 seconds
(2/16) Running aws-sdk-java tests ... done in 1 seconds
(3/16) Running aws-sdk-php tests ... done in 41 seconds
(4/16) Running aws-sdk-ruby tests ... done in 2 seconds
(5/16) Running awscli tests ... done in 2 minutes and 8 seconds
(6/16) Running healthcheck tests ... done in 1 seconds
(7/16) Running mc tests ... done in 39 seconds
(8/16) Running minio-dotnet tests ... done in 34 seconds
(9/16) Running minio-go tests ... FAILED in 44 seconds
{
  "args": {
    "bucketName": "minio-go-test-21vumx3hr3nn1ocs",
    "objectName": "m04mlqojibh5d9ruzwejscwnlc0yte",
    "sse": {}
  },
  "duration": 22,
  "error": "Server side encryption specified but KMS is not configured",
  "function": "PutEncryptedObject(bucketName, objectName, reader, sse)",
  "message": "PutEncryptedObject failed",
  "name": "minio-go: testSSES3EncryptionPutGet",
  "status": "FAIL"
}
(9/16) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "setBucketEncryption()",
  "duration": 31,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidArgument, message = Server side encryption specified but KMS is not configured, bucketName = minio-java-test-1n7c5ge, objectName = null, resource = /minio-java-test-1n7c5ge, requestId = 168876C6F62999FB, hostId = 2fcfc421-e827-487b-b7ba-09c6d013ecfa)\nrequest={method=PUT, url=http://minio-c3.minio.io:30560/minio-java-test-1n7c5ge?encryption=, headers=Host: minio-c3.minio.io:30560\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.2.1\nContent-MD5: 5bZurju84n5lUGMQ6mrGXQ==\nx-amz-content-sha256: df4e3e6cf7a0b4af67ba58f41eaa596d54579eea2ed2f993da09600f59413711\nx-amz-date: 20210614T134422Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210614/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 344\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 168876C6F62999FB\nX-Xss-Protection: 1; mode=block\nDate: Mon, 14 Jun 2021 13:44:22 GMT\n}\n >>> [io.minio.S3Base.execute(S3Base.java:667), io.minio.S3Base.execute(S3Base.java:460), io.minio.S3Base.executePut(S3Base.java:794), io.minio.MinioClient.setBucketEncryption(MinioClient.java:2343), FunctionalTest.setBucketEncryption(FunctionalTest.java:3223), FunctionalTest.runBucketTests(FunctionalTest.java:3656), FunctionalTest.runTests(FunctionalTest.java:3719), FunctionalTest.main(FunctionalTest.java:3864)]"
}
(9/16) Running minio-js tests ... done in 42 seconds
(10/16) Running minio-py tests ... done in 1 minutes and 10 seconds
(11/16) Running s3cmd tests ... done in 13 seconds
(12/16) Running s3select tests ... done in 2 seconds
(13/16) Running security tests ... done in 0 seconds
(14/16) Running versioning tests ... done in 0 seconds

Executed 14 out of 16 tests successfully.
```
<a name="ee652dae-c9ba-4d54-be60-15967f2187b1"></a>
### 12502-c9f1a61/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:31905
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 83293209a400:/mint/log /tmp/mint-logs'

(1/16) Running aws-sdk-go tests ... done in 1 seconds
(2/16) Running aws-sdk-java tests ... done in 1 seconds
(3/16) Running aws-sdk-php tests ... done in 43 seconds
(4/16) Running aws-sdk-ruby tests ... done in 3 seconds
(5/16) Running awscli tests ... done in 2 minutes and 23 seconds
(6/16) Running healthcheck tests ... done in 0 seconds
(7/16) Running mc tests ... done in 1 minutes and 13 seconds
(8/16) Running minio-dotnet tests ... done in 39 seconds
(9/16) Running minio-go tests ... FAILED in 1 minutes and 17 seconds
{
  "args": {
    "bucketName": "minio-go-test-akjuhjn0bpbdjjxb",
    "objectName": "r16s12qs6sxiqnu3rls49dr2lnr5y2",
    "sse": {}
  },
  "duration": 57,
  "error": "Server side encryption specified but KMS is not configured",
  "function": "PutEncryptedObject(bucketName, objectName, reader, sse)",
  "message": "PutEncryptedObject failed",
  "name": "minio-go: testSSES3EncryptionPutGet",
  "status": "FAIL"
}
(9/16) Running minio-java tests ... FAILED in 3 seconds
{
  "name": "minio-java",
  "function": "setBucketEncryption()",
  "duration": 94,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InvalidArgument, message = Server side encryption specified but KMS is not configured, bucketName = minio-java-test-u0t514, objectName = null, resource = /minio-java-test-u0t514, requestId = 168876DC578F69EB, hostId = 60f0def4-0f5f-4a21-9071-b885c3c246fd)\nrequest={method=PUT, url=http://minio-c3.minio.io:31905/minio-java-test-u0t514?encryption=, headers=Host: minio-c3.minio.io:31905\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.2.1\nContent-MD5: 5bZurju84n5lUGMQ6mrGXQ==\nx-amz-content-sha256: df4e3e6cf7a0b4af67ba58f41eaa596d54579eea2ed2f993da09600f59413711\nx-amz-date: 20210614T134554Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210614/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=400, headers=Accept-Ranges: bytes\nContent-Length: 342\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 168876DC578F69EB\nX-Xss-Protection: 1; mode=block\nDate: Mon, 14 Jun 2021 13:45:54 GMT\n}\n >>> [io.minio.S3Base.execute(S3Base.java:667), io.minio.S3Base.execute(S3Base.java:460), io.minio.S3Base.executePut(S3Base.java:794), io.minio.MinioClient.setBucketEncryption(MinioClient.java:2343), FunctionalTest.setBucketEncryption(FunctionalTest.java:3223), FunctionalTest.runBucketTests(FunctionalTest.java:3656), FunctionalTest.runTests(FunctionalTest.java:3719), FunctionalTest.main(FunctionalTest.java:3864)]"
}
(9/16) Running minio-js tests ... done in 52 seconds
(10/16) Running minio-py tests ... done in 2 minutes and 3 seconds
(11/16) Running s3cmd tests ... done in 20 seconds
(12/16) Running s3select tests ... done in 6 seconds
(13/16) Running security tests ... done in 0 seconds
(14/16) Running versioning tests ... done in 3 minutes and 6 seconds

Executed 14 out of 16 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/erasure-server-pool.go
+++ b/cmd/erasure-server-pool.go
@@ -650,19 +650,22 @@ func (z *erasureServerPools) GetObjectNInfo(ctx context.Context, bucket, object 
 		unlockOnDefer = true
 	}
 
-	errs := make([]error, len(z.serverPools))
-	grs := make([]*GetObjectReader, len(z.serverPools))
-
 	lockType = noLock // do not take locks at lower levels
 	checkPrecondFn := opts.CheckPrecondFn
 	opts.CheckPrecondFn = nil
+	results := make([]struct {
+		zIdx int
+		gr   *GetObjectReader
+		err  error
+	}, len(z.serverPools))
 
 	var wg sync.WaitGroup
 	for i, pool := range z.serverPools {
 		wg.Add(1)
 		go func(i int, pool *erasureSets) {
 			defer wg.Done()
-			grs[i], errs[i] = pool.GetObjectNInfo(ctx, bucket, object, rs, h, lockType, opts)
+			results[i].zIdx = i
+			results[i].gr, results[i].err = pool.GetObjectNInfo(ctx, bucket, object, rs, h, lockType, opts)
 		}(i, pool)
 	}
 	wg.Wait()
@@ -671,61 +674,67 @@ func (z *erasureServerPools) GetObjectNInfo(ctx context.Context, bucket, object 
 	// this is a defensive change to handle any duplicate
 	// content that may have been created, we always serve
 	// the latest object.
-	less := func(i, j int) bool {
-		var (
-			mtime1 time.Time
-			mtime2 time.Time
-		)
-		if grs[i] != nil {
-			mtime1 = grs[i].ObjInfo.ModTime
-		}
-		if grs[j] != nil {
-			mtime2 = grs[j].ObjInfo.ModTime
-		}
-		return mtime1.After(mtime2)
-	}
+	sort.Slice(results, func(i, j int) bool {
+		var mtimeI, mtimeJ time.Time
 
-	// Must be a stable sort for both slices to be sorted the same.
-	sort.SliceStable(errs, less)
-	sort.SliceStable(grs, less)
+		if results[i].gr != nil {
+			mtimeI = results[i].gr.ObjInfo.ModTime
+		}
+		if results[j].gr != nil {
+			mtimeJ = results[j].gr.ObjInfo.ModTime
+		}
+		// On tiebreaks, choose the earliest.
+		if mtimeI == mtimeJ {
+			return results[i].zIdx < results[j].zIdx
+		}
+		return mtimeI.After(mtimeJ)
+	})
 
 	var found = -1
-	for i, err := range errs {
-		if err == nil {
+	for i, res := range results {
+		if res.err == nil {
+			// Select earliest result
+			if res.gr == nil {
+				panic(err)
+			}
 			found = i
 			break
 		}
-		if !isErrObjectNotFound(err) && !isErrVersionNotFound(err) {
-			for _, grr := range grs {
-				if grr != nil {
-					grr.Close()
+		if !isErrObjectNotFound(res.err) && !isErrVersionNotFound(res.err) {
+			for _, res := range results {
+				if res.gr != nil {
+					res.gr.Close()
 				}
 			}
-			return gr, err
+			return nil, err
 		}
 	}
 
-	if found >= 0 {
-		if checkPrecondFn != nil && checkPrecondFn(grs[found].ObjInfo) {
-			for _, grr := range grs {
-				grr.Close()
-			}
-			return nil, PreConditionFailed{}
+	if found < 0 {
+		object = decodeDirObject(object)
+		if opts.VersionID != "" {
+			return gr, VersionNotFound{Bucket: bucket, Object: object, VersionID: opts.VersionID}
 		}
-		for i, grr := range grs {
-			if i == found {
-				continue
-			}
-			grr.Close()
-		}
-		return grs[found], nil
+		return gr, ObjectNotFound{Bucket: bucket, Object: object}
 	}
 
-	object = decodeDirObject(object)
-	if opts.VersionID != "" {
-		return gr, VersionNotFound{Bucket: bucket, Object: object, VersionID: opts.VersionID}
+	// Check preconditions.
+	if checkPrecondFn != nil && checkPrecondFn(results[found].gr.ObjInfo) {
+		for _, res := range results {
+			res.gr.Close()
+		}
+		return nil, PreConditionFailed{}
 	}
-	return gr, ObjectNotFound{Bucket: bucket, Object: object}
+
+	// Close all others
+	for i, res := range results {
+		if i == found {
+			continue
+		}
+		res.gr.Close()
+	}
+
+	return results[found].gr, nil
 }
 
 func (z *erasureServerPools) GetObjectInfo(ctx context.Context, bucket, object string, opts ObjectOptions) (objInfo ObjectInfo, err error) {
@@ -747,15 +756,19 @@ func (z *erasureServerPools) GetObjectInfo(ctx context.Context, bucket, object s
 	}
 	defer lk.RUnlock()
 
-	errs := make([]error, len(z.serverPools))
-	objInfos := make([]ObjectInfo, len(z.serverPools))
+	results := make([]struct {
+		zIdx int
+		oi   ObjectInfo
+		err  error
+	}, len(z.serverPools))
 	opts.NoLock = true // avoid taking locks at lower levels for multi-pool setups.
 	var wg sync.WaitGroup
 	for i, pool := range z.serverPools {
 		wg.Add(1)
 		go func(i int, pool *erasureSets) {
 			defer wg.Done()
-			objInfos[i], errs[i] = pool.GetObjectInfo(ctx, bucket, object, opts)
+			results[i].zIdx = i
+			results[i].oi, results[i].err = pool.GetObjectInfo(ctx, bucket, object, opts)
 		}(i, pool)
 	}
 	wg.Wait()
@@ -764,30 +777,25 @@ func (z *erasureServerPools) GetObjectInfo(ctx context.Context, bucket, object s
 	// this is a defensive change to handle any duplicate
 	// content that may have been created, we always serve
 	// the latest object.
-	less := func(i, j int) bool {
-		mtime1 := objInfos[i].ModTime
-		mtime2 := objInfos[j].ModTime
-		return mtime1.After(mtime2)
-	}
-	// Must be a stable sort for both slices to be sorted the same.
-	sort.SliceStable(errs, less)
-	sort.SliceStable(objInfos, less)
-
-	var found = -1
-	for i, err := range errs {
+	sort.Slice(results, func(i, j int) bool {
+		a, b := results[i], results[j]
+		if a.oi.ModTime.Equal(b.oi.ModTime) {
+			// On tiebreak, select the lowest zone index.
+			return a.zIdx < b.zIdx
+		}
+		return a.oi.ModTime.After(b.oi.ModTime)
+	})
+	for _, res := range results {
+		// Return first found.
+		err := res.err
 		if err == nil {
-			found = i
-			break
+			return res.oi, nil
 		}
 		if !isErrObjectNotFound(err) && !isErrVersionNotFound(err) {
 			// some errors such as MethodNotAllowed for delete marker
 			// should be returned upwards.
-			return objInfos[i], err
+			return res.oi, err
 		}
-	}
-
-	if found >= 0 {
-		return objInfos[found], nil
 	}
 
 	object = decodeDirObject(object)

--- a/cmd/erasure-server-pool.go
+++ b/cmd/erasure-server-pool.go
@@ -684,7 +684,7 @@ func (z *erasureServerPools) GetObjectNInfo(ctx context.Context, bucket, object 
 			mtimeJ = results[j].gr.ObjInfo.ModTime
 		}
 		// On tiebreaks, choose the earliest.
-		if mtimeI == mtimeJ {
+		if mtimeI.Equal(mtimeJ) {
 			return results[i].zIdx < results[j].zIdx
 		}
 		return mtimeI.After(mtimeJ)

